modin/engines/base/io/file_reader.py

```diff
@@ -23,6 +24,32 @@ class FileReader:
     frame_partition_cls = None
     query_compiler_cls = None
 
+    @classmethod
+    def read(cls, *args, **kwargs):
+        query_compiler = cls._read(*args, **kwargs)
+        # TODO (devin-petersohn): Make this section more general for non-pandas kernel
```

Comment:
Probably we need to open an issue for this one. At least omnisci will need this to be changed probably, but we can fix in a later PR.

Reply:
Ok #1504